staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go

```diff
@@ -1219,6 +1222,190 @@ func RunSendInitialEventsBackwardCompatibility(ctx context.Context, t *testing.T
 	w.Stop()
 }
 
+func RunWatchSemantics(ctx context.Context, t *testing.T, newStorage func() (storage.Interface, func())) {
```

Comment:
let's discuss only this test for now.

Reply:
i'll move it to a separate PR.

Reply:
Overall status for etcd impl:
![Screenshot 2023-09-06 at 14 12 45](https://github.com/kubernetes/kubernetes/assets/21304151/bfdc5a0c-3cd8-4ba1-83f9-26d3637229d5)

Overall status for the cache:
![Screenshot 2023-09-06 at 14 15 40](https://github.com/kubernetes/kubernetes/assets/21304151/fdce37b8-15ba-484f-ac20-771d5bd8c80e)

Reply:
In the end only 2 cases seems to diverge.